File: doc/examples/workspace_taskapp/taskapp-api/src/task/app/TaskService.java
package task.app;

import java.util.List;

import byps.BRemote;
import byps.RemoteException;

public interface TaskService extends BRemote {

	void addTask(TaskInfo task) throws RemoteException;
	
	void removeTask(long taskId) throws RemoteException;
	
	List<TaskInfo> getTasks() throws RemoteException;
	
}

File: doc/examples/workspace_taskapp/taskapp-ser/src-cs/task/app/TaskService.cs
using System;
using System.Collections.Generic;
using byps;

namespace task.app
{
	
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface TaskService : BRemote {
	
		void AddTask(TaskInfo task);
		void AddTask(TaskInfo task, BAsyncResult<Object> asyncResult) ;
		
		void RemoveTask(long taskId);
		void RemoveTask(long taskId, BAsyncResult<Object> asyncResult) ;
		
		IList<TaskInfo> GetTasks();
		void GetTasks(BAsyncResult<IList<TaskInfo>> asyncResult) ;
		
		
	}
}  // end namespace
